app/controllers/project_relationships_controller.rb

```diff
@@ -0,0 +1,81 @@
+# RESTfully manages creating/deleting `ProjectRelationship`s
+class ProjectRelationshipsController < ApplicationController
+  load_and_authorize_resource :project
+  load_and_authorize_resource :team, through: :project, singleton: true
+  load_and_authorize_resource through: :project, only: %i[create destroy]
+
+  def index
+    authorize!(:read, ProjectRelationship)
+
+    @projects = @team.projects.
+                accessible_by(current_ability).
+                where.not(id: @project).
+                search(search_params).
+                order_by_reference.
+                includes(:project_type, :owner, :current_state).
+                paginate(page: params[:page], per_page: 10)
+  end
+
+  def create
+    @other_project = @team.projects.find_by(id: resource_params[:right_project_id])
+
+    @project_relationship.assign_attributes(
+      left_project:  @project,
+      right_project: @other_project
+    )
+
+    if @project_relationship.save
+      respond_to do |format|
+        format.html { redirect_to after_action_path, notice: t('.success') }
+        format.js
+      end
+    else
+      respond_to do |format|
+        format.html { redirect_to after_action_path, alert: t('.failure') }
+        format.js
+      end
+    end
+  end
+
+  def destroy
+    @other_project = @project_relationship.projects.detect { |project| project != @project }
```

Comment:
More of a question than anything - would where.not(id: @project.id) be more efficient?

Reply:
Good spot, and you're absolutely right. 

It's still one DB hit, but the difference is in loading one record into memory rather than two. There are times when relying on the preloaded/cached association would be beneficial but this isn't one of those instances :D 

This code was a remnant from when associations weren't as fleshed out as they are now and `project_relationships.projects` was just an enumerable wrapper around the two `belongs_to` associations.

I'll push a commit with your suggested change :+1: